pkg/controller/snapshot_controller.go

```diff
@@ -421,13 +421,65 @@ func (ctrl *csiSnapshotController) checkandBindSnapshotContent(snapshot *crdv1.V
 	return nil
 }
 
+func (ctrl *csiSnapshotController) prepareCreateSnapshotInput(snapshot *crdv1.VolumeSnapshot) (*crdv1.VolumeSnapshotClass, *v1.PersistentVolume, string, map[string]string, error) {
+	className := snapshot.Spec.VolumeSnapshotClassName
+	glog.V(5).Infof("prepareCreateSnapshotInput [%s]: VolumeSnapshotClassName [%s]", snapshot.Name, *className)
+	var class *crdv1.VolumeSnapshotClass
+	var err error
+	if className != nil {
+		class, err = ctrl.GetSnapshotClass(*className)
+		if err != nil {
+			glog.Errorf("prepareCreateSnapshotInput failed to getClassFromVolumeSnapshot %s", err)
+			return nil, nil, "", nil, err
+		}
+	} else {
+		glog.Errorf("failed to prepareCreateSnapshotInput %s without a snapshot class", snapshot.Name)
+		return nil, nil, "", nil, fmt.Errorf("failed to take snapshot %s without a snapshot class", snapshot.Name)
+	}
+
+	volume, err := ctrl.getVolumeFromVolumeSnapshot(snapshot)
+	if err != nil {
+		glog.Errorf("prepareCreateSnapshotInput failed to get PersistentVolume object [%s]: Error: [%#v]", snapshot.Name, err)
+		return nil, nil, "", nil, err
+	}
+
+	// Create VolumeSnapshotContent name
+	contentName := GetSnapshotContentNameForSnapshot(snapshot)
+
+	// Resolve snapshotting secret credentials.
+	snapshotterSecretRef, err := GetSecretReference(class.Parameters, contentName, snapshot)
+	if err != nil {
+		return nil, nil, "", nil, err
+	}
+	snapshotterCredentials, err := GetCredentials(ctrl.client, snapshotterSecretRef)
+	if err != nil {
+		return nil, nil, "", nil, err
+	}
+
+	return class, volume, contentName, snapshotterCredentials, nil
+}
+
 func (ctrl *csiSnapshotController) checkandUpdateBoundSnapshotStatusOperation(snapshot *crdv1.VolumeSnapshot, content *crdv1.VolumeSnapshotContent) (*crdv1.VolumeSnapshot, error) {
-	status, _, size, err := ctrl.handler.GetSnapshotStatus(content)
+	var err error
+	var timestamp int64 = 0
+	var size int64 = 0
+	var readyToUse bool = false
+	class, volume, _, snapshotterCredentials, err := ctrl.prepareCreateSnapshotInput(snapshot)
 	if err != nil {
-		return nil, fmt.Errorf("failed to check snapshot status %s with error %v", snapshot.Name, err)
+		return nil, fmt.Errorf("failed to prepare input parameters to call create snapshot of the volume, %s: %q", volume.Name, err)
 	}
-	timestamp := time.Now().UnixNano()
-	newSnapshot, err := ctrl.updateSnapshotStatus(snapshot, status, timestamp, size, IsSnapshotBound(snapshot, content))
+	driverName, snapshotID, timestamp, size, readyToUse, err := ctrl.handler.CreateSnapshot(snapshot, volume, class.Parameters, snapshotterCredentials)
+	if err != nil {
+		// Don't return here. Update status by calling updateSnapshotStatus below
+		glog.Errorf("checkandUpdateBoundSnapshotStatusOperation: failed to create snapshot %q", err)
```

Comment:
here we call `CreateSnapshot` is to query whether the snapshot is ready to use, can we update the message here to `
checkandUpdateBoundSnapshotStatusOperation: failed to call create snapshot for check whether the snapshot is ready to use %q`